downstairs/src/lib.rs

```diff
@@ -1018,118 +1018,100 @@ async fn do_work_task(
-                    let m = ads
-                        .lock()
-                        .await
-                        .do_work(upstairs_connection, job_id)
-                        .await?;
+            // If the job's dependencies aren't met, then keep going
+            let Some(job_id) = job_id else {
+                continue;
+            };
 
-                    // If this is a repair job, and that repair failed, we
-                    // can do no more work on this downstairs and should
-                    // force everything to come down before more work arrives.
-                    //
-                    // However, we can respond to the upstairs with the failed
-                    // result, and let the upstairs take action that will
-                    // allow it to abort the repair and continue working in
-                    // some degraded state.
-                    let mut abort_needed = false;
-                    if let Some(m) = m {
-                        abort_needed = check_message_for_abort(&m);
-
-                        if let Some(error) = m.err() {
-                            resp_tx
-                                .send(Message::ErrorReport {
-                                    upstairs_id: upstairs_connection
-                                        .upstairs_id,
-                                    session_id: upstairs_connection.session_id,
-                                    job_id: new_id,
-                                    error: error.clone(),
-                                })
-                                .await?;
-
-                            // If the job errored, do not consider it completed.
-                            // Retry it.
-                            repeat_work.push(new_id);
-                        } else {
-                            // The job completed successfully, so inform the
-                            // Upstairs
-
-                            ads.lock().await.complete_work_stat(
-                                upstairs_connection,
-                                &m,
-                                job_id,
-                            )?;
-
-                            // Notify the upstairs before completing work, which
-                            // consumes the message (so we'll check whether it's
-                            // a FlushAck beforehand)
-                            let is_flush =
-                                matches!(m, Message::FlushAck { .. });
-                            resp_tx.send(m).await?;
-
-                            ads.lock()
-                                .await
-                                .complete_work_inner(
-                                    upstairs_connection,
-                                    job_id,
-                                    is_flush,
-                                )
-                                .await?;
+            cdt::work__process!(|| job_id.0);
```

Comment:
I thought maybe this probe was in a bad spot, as we could hit this, then
fal out of the loop if some other downstairs was promoted at line 1057,
but, if that really is the case, then the whole accounting is meaningless 
as we should be shutting this task down with the other upstairs takes over.

Also, this issue was here before you refactored.